pkg/apis/pipeline/v1/container_types_test.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2023 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"reflect"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+func TestSidecar_ToK8sContainer(t *testing.T) {
+	type fields struct {
+		Name                     string
+		Image                    string
+		Command                  []string
+		Args                     []string
+		WorkingDir               string
+		Ports                    []v1.ContainerPort
+		EnvFrom                  []v1.EnvFromSource
+		Env                      []v1.EnvVar
+		ComputeResources         v1.ResourceRequirements
+		VolumeMounts             []v1.VolumeMount
+		VolumeDevices            []v1.VolumeDevice
+		LivenessProbe            *v1.Probe
+		ReadinessProbe           *v1.Probe
+		StartupProbe             *v1.Probe
+		Lifecycle                *v1.Lifecycle
+		TerminationMessagePath   string
+		TerminationMessagePolicy v1.TerminationMessagePolicy
+		ImagePullPolicy          v1.PullPolicy
+		SecurityContext          *v1.SecurityContext
+		Stdin                    bool
+		StdinOnce                bool
+		TTY                      bool
+		Script                   string
+		Workspaces               []WorkspaceUsage
+	}
+	tests := []struct {
+		name   string
+		fields fields
```

Comment:
could this be replaced with a sidecar?
```suggestion
		sidecar *Sidecar
```